nomad/mock/alloc.go

```diff
@@ -87,14 +87,19 @@ func Alloc() *structs.Allocation {
 }
 
 func MinAlloc() *structs.Allocation {
-	job := MinJob()
+	return MinAllocForJob(MinJob())
+}
+
+func MinAllocForJob(job *structs.Job) *structs.Allocation {
 	group := job.TaskGroups[0]
 	task := group.Tasks[0]
```

Comment:
accepting any ol' Job opens up the possibility of panics if a caller provides a job with no groups or tasks, but the stack trace will not be very mysterious to a caller holding it wrong.